cmd/secret/cmd/root.go

```diff
@@ -100,31 +108,48 @@ func (r *root) setCmd() *cobra.Command {
 	var value string
 	var path string
 	var redisURL string
+	var postgresURL string
+	var migration string
 	var setCmd = &cobra.Command{
 		Use:   "set",
 		Short: "Saves data to the specified storage in encrypted form",
 		Long:  "it takes keys and a value from user and saves value in encrypted manner in specified storage",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			var ds secretApi.DataSaver
 			var cr = crypto.NewCryptographer([]byte(cipherKey))
-			rdb := redis.NewClient(&redis.Options{Addr: redisURL, Password: "", DB: 0})
-			err := rdb.Ping(r.cmd.Context()).Err()
-			if err != nil {
-				return fmt.Errorf("redis db is not reachable:  %w", err)
-			}
-
+			logger := r.logger.Named("set cmd")
 			switch {
 			case redisURL != "":
+				rdb := redis.NewClient(&redis.Options{Addr: redisURL, Password: "", DB: 0})
+				err := rdb.Ping(r.cmd.Context()).Err()
+				if err != nil {
+					return fmt.Errorf("redis db is not reachable:  %w", err)
+				}
 				ds = storage.NewRedisVault(rdb)
+			case postgresURL != "":
+				err := migrateUp(postgresURL, migration)
+				if err != nil {
+					if errors.Is(err, migrate.ErrNoChange) {
+						logger.Infof("can't migrate db:  %s", err)
+					} else {
+						return fmt.Errorf("migrate error :  %w", err)
+					}
+				}
+				pdb, err := sqlx.ConnectContext(r.cmd.Context(), "postgres", postgresURL)
```

Comment:
If `pdb` was created successfully it establishes connections that should be closed after their usage is done. Could use `defer` to perform disconnect.

The same is applicable for redis db.

Use disconnect/close for every cli command usage.